torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1392,77 +1392,8 @@ Example::
-      .def_readwrite(
-          "total_parameter_size_bytes",
-          &c10::DDPLoggingData::total_parameter_size_bytes)
-      .def_readwrite(
-          "num_parameter_tensors", &c10::DDPLoggingData::num_parameter_tensors)
-      .def_readwrite("dtypes", &c10::DDPLoggingData::dtypes)
-      .def_readwrite("bucket_sizes", &c10::DDPLoggingData::bucket_sizes)
-      .def_readwrite("master_port", &c10::DDPLoggingData::master_port)
-      .def_readwrite("master_addr", &c10::DDPLoggingData::master_addr)
-      .def_readwrite(
-          "cuda_visible_devices", &c10::DDPLoggingData::cuda_visible_devices)
-      .def_readwrite(
-          "gloo_socket_ifname", &c10::DDPLoggingData::gloo_socket_ifname)
-      .def_readwrite(
-          "gloo_device_transport", &c10::DDPLoggingData::gloo_device_transport)
-      .def_readwrite(
-          "nccl_socket_ifname", &c10::DDPLoggingData::nccl_socket_ifname)
-      .def_readwrite(
-          "nccl_blocking_wait", &c10::DDPLoggingData::nccl_blocking_wait)
-      .def_readwrite(
-          "nccl_async_error_handling",
-          &c10::DDPLoggingData::nccl_async_error_handling)
-      .def_readwrite("nccl_debug", &c10::DDPLoggingData::nccl_debug)
-      .def_readwrite("nccl_nthreads", &c10::DDPLoggingData::nccl_nthreads)
-      .def_readwrite("nccl_ib_timeout", &c10::DDPLoggingData::nccl_ib_timeout)
-      .def_readwrite(
-          "unused_parameter_size", &c10::DDPLoggingData::unused_parameter_size)
-      .def_readwrite(
-          "has_rebuilt_buckets", &c10::DDPLoggingData::has_rebuilt_buckets)
-      .def_readwrite(
-          "rebuilt_bucket_sizes", &c10::DDPLoggingData::rebuilt_bucket_sizes)
-      .def_readwrite(
-          "avg_forward_compute_time",
-          &c10::DDPLoggingData::avg_forward_compute_time)
-      .def_readwrite(
-          "avg_backward_compute_time",
-          &c10::DDPLoggingData::avg_backward_compute_time)
-      .def_readwrite(
-          "avg_backward_comm_time",
-          &c10::DDPLoggingData::avg_backward_comm_time)
-      .def_readwrite(
-          "avg_backward_compute_comm_overlap_time",
-          &c10::DDPLoggingData::avg_backward_compute_comm_overlap_time)
-      .def_readwrite("comm_hook", &c10::DDPLoggingData::comm_hook)
-      .def_readwrite("join_uneven_inputs", &c10::DDPLoggingData::join_uneven_inputs)
-      .def_readwrite(
-          "forward_compute_time", &c10::DDPLoggingData::forward_compute_time)
-      .def_readwrite(
-          "backward_compute_time", &c10::DDPLoggingData::backward_compute_time)
-      .def_readwrite(
-          "backward_comm_time", &c10::DDPLoggingData::backward_comm_time)
-      .def_readwrite(
-          "backward_compute_comm_overlap_time",
-          &c10::DDPLoggingData::backward_compute_comm_overlap_time)
-      .def_readwrite("is_multi_device_module", &c10::DDPLoggingData::is_multi_device_module);
+      .def_readwrite("strs_map", &c10::DDPLoggingData::strs_map)
```

Comment:
You can also consider keeping a single map, where the value type is `c10::either<int, std::string>`.

Therefore, you can even just aliasing this class instead of creating a struct.
`using DDPLogingData = std::unordered_map<std::string, c10::either<int, std::string>>;`

Reply:
You may also consider std::map instead. If the users don't know the field names and want to print them out, this can give a nicer and deterministic order. I don't think we need to care too much about the search performance here, since the map size is relatively small.

Reply:
good suggestions, thanks!

Reply:
> Optional: You can also consider keeping a single map, where the value type is `c10::either<int, std::string>`.
> 
> Therefore, you can even just aliasing this class instead of creating a struct.
> `using DDPLogingData = std::unordered_map<std::string, c10::either<int, std::string>>;`

just had second thoughts, c10::either will be hard to let us add more types in addition to int and string in the future. So I will keep current implementation for now